bravado/exception.py

```diff
@@ -230,3 +236,7 @@ class HTTPNotExtended(HTTPServerError):
 
 class HTTPNetworkAuthenticationRequired(HTTPServerError):
     status_code = 511
+
+
+class BravadoTimeoutError(base_exception):
```

Comment:
This is a backwards-incompatible change, and changing the type of timeout exception fido threw when calling `result()` caused quite a bit of pain until we found a backwards-compatible solution. Maybe we could do the same here?

1. Define client-specific timeout exceptions like RequestsTimeoutError, FidoTimeoutError, AsyncioTimeoutError
1. Make them inherit from both BravadoTimeoutError and their "native" timeout exception, like fido.exceptions.HTTPTimeoutError, requests.exceptions.ReadTimeout or aiohttp.ServerTimeoutError
1. Do the same thing for connection errors (we have the same issue there)

While it'd be ideal if we could implement this unification only in bravado, I'm perfectly fine with extending the API contract between bravado and the HTTP client to require the client to define their own connection and timeout exceptions that need to be subclasses of bravado ones.

I also think there is value in requiring HTTP clients to behave like that: We can be certain that the bravado library with this change will throw consistent timeout and connection exceptions. We'll just require new fido and bravado-asyncio versions to be used with it. I.e. I probably wouldn't make @reraise_timeout_errors fail gracefully.

Reply:
I don't see how this change could be back incompatible.
``raraise_timeout_errors`` decorator makes sure to raise exceptions dynamically generated if not defined in ``FutureAdapter.__timeout_error_type`` that are extending the timeout exceptions and ``BravadoTimeoutException`` ([Reference](https://github.com/Yelp/bravado/pull/321/files#diff-dfa77bc847cf31d43435498de902fe9bR47)).
This means that you can continue to catch timeout exceptions how you were used to (using HTTP client specific timeout exceptions) and using the new base class for bravado timeout exceptions.

About connections errors, I was thinking to unify this type of errors too but I was having issues in writing a valid integration test for fido client so I decided to postpone it in a separate PR

Reply:
Sounds good, I didn't look closely enough at the decorator implementation initially.